scripts/quick-helm-install.sh

```diff
@@ -0,0 +1,111 @@
+        echo "-s, --small               reduce StackRox resource requirements for small clusters"
+        exit 0
+        ;;
+    -s|--small)
+        SMALL_INSTALL=true
+        ;;
+esac
+
+echo "Adding the stackrox/helm-charts/opensource repository to Helm."
+
+helm repo add stackrox https://raw.githubusercontent.com/stackrox/helm-charts/main/opensource/
+
+echo "Generating STACKROX_ADMIN_PASSWORD"
+
+STACKROX_ADMIN_PASSWORD="$(openssl rand -base64 20 | tr -d '/=+')"
+
+echo "Installing stackrox-central-services"
+
+helm install -n stackrox --create-namespace stackrox-central-services stackrox/stackrox-central-services --set central.adminPassword.value="${STACKROX_ADMIN_PASSWORD}"
+
+if [[ "$SMALL_INSTALL" == "true" ]]; then
+    echo "Upgrading stackrox to a small install"
+    helm upgrade -n stackrox stackrox-central-services stackrox/stackrox-central-services \
+      --set central.resources.requests.memory=1Gi \
+      --set central.resources.requests.cpu=1 \
+      --set central.resources.limits.memory=4Gi \
+      --set central.resources.limits.cpu=1 \
+      --set scanner.autoscaling.disable=true \
+      --set scanner.replicas=1 \
+      --set scanner.resources.requests.memory=500Mi \
+      --set scanner.resources.requests.cpu=500m \
+      --set scanner.resources.limits.memory=2500Mi \
+      --set scanner.resources.limits.cpu=2000m
+fi
+
+kubectl -n stackrox rollout status deploy/central --timeout=3m
+
+echo "Forwarding ports to central"
+
+kubectl -n stackrox port-forward deploy/central --pod-running-timeout=1m0s 8000:8443 > /dev/null 2>&1 &
+
+echo "Generating an init bundle with shared secrets"
+
+kubectl -n stackrox exec deploy/central -- roxctl --insecure-skip-tls-verify \
+  --password "${STACKROX_ADMIN_PASSWORD}" \
+  central init-bundles generate stackrox-init-bundle --output - 1> stackrox-init-bundle.yaml
+
+installflags=('-f' 'stackrox-init-bundle.yaml' '--set' 'clusterName="my-secured-cluster"')
+if [[ "$SMALL_INSTALL" == "true" ]]; then
+    installflags+=('--set' 'sensor.resources.requests.memory=500Mi')
+    installflags+=('--set' 'sensor.resources.requests.cpu=500m')
+    installflags+=('--set' 'sensor.resources.limits.memory=500Mi')
+    installflags+=('--set' 'sensor.resources.limits.cpu=500m')
+    echo "Installing the first small secured cluster"
+else
+    echo "Installing the first secured cluster"
+fi
+
+helm install -n stackrox stackrox-secured-cluster-services stackrox/stackrox-secured-cluster-services "${installflags[@]}"
+
+echo "Logging into StackRox in the browser"
+
+_output="$(mktemp)"
+if curl -sSkf -u "admin:${STACKROX_ADMIN_PASSWORD}" --output "${_output}" -w '%{redirect_url}' "https://localhost:8000/sso/providers/basic/4df1b98c-24ed-4073-a9ad-356aec6bb62d/challenge?micro_ts=0"; then
+    target_url="$(cat "${_output}")"
+    if [[ -n "$target_url" ]]; then
+        auth_error="$(sed -E '/^.*#error=([^&]+)(&.*)?$/!d;s//\1/' <<<"$target_url" | tr '+' ' ')"
```

Comment:
```bash
sed -E '/^.*#error=([^&]+)(&.*)?$/!d;s//\1/' <<<"$target_url" | tr '+' ' ')"
```
What happens here?

Reply:
sed -E interprets extended RegEx. `$target_url` is a URL to login with the credentials `admin:${STACKROX_ADMIN_PASSWORD}` queried in the previous step, but sometimes that step fails.  The RegEx `'/^.*#error=([^&]+)(&.*)?$/!d;s//\1/'` checks if there's an error within `$target_url`, in which case the next if condition 
`if [[ -z "$auth_error" ]]; then` will fail and output `echo "Authentication error: ${auth_error}" 1>&2` 

This sed command is also taken from the logmein workflow script ([here](https://github.com/stackrox/workflow/blob/master/scripts/runtime/logmein.sh), line 58)